From 5d483adfe54861a5d99bb409a9a9b934796aef66 Mon Sep 17 00:00:00 2001
From: David Francis <David.Francis@amd.com>
Date: Mon, 4 Mar 2019 10:31:31 -0500
Subject: [PATCH 1880/2940] drm/amd/display: Update ABM crtc state on
 non-modeset

[Why]
Somewhere in the atomic check reshuffle ABM got lost.
ABM is a crtc property (copied from a connector property).
It can change without a modeset, just like underscan.

[How]
In the skip_modeset branch of atomic check crtc updates,
copy over the abm property.

Change-Id: I42f4b938749e4bfbc1637fa8cb378f5b78c9b0dd
Signed-off-by: David Francis <David.Francis@amd.com>
Reviewed-by: Nicholas Kazlauskas <Nicholas.Kazlauskas@amd.com>
Acked-by: Bhawanpreet Lakha <Bhawanpreet.Lakha@amd.com>
Signed-off-by: Chaudhary Amit Kumar <Chaudharyamit.Kumar@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index 6363cf04f41d..50d642bd304c 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -6370,6 +6370,10 @@ static int dm_update_crtc_state(struct amdgpu_display_manager *dm,
         if (is_scaling_state_different(dm_old_conn_state, dm_new_conn_state))
                 update_stream_scaling_settings(
                         &new_crtc_state->mode, dm_new_conn_state, dm_new_crtc_state->stream);
+
+	/* ABM settings */
+        dm_new_crtc_state->abm_level = dm_new_conn_state->abm_level;
+
         /*
          * Color management settings. We also update color properties
          * when a modeset is needed, to ensure it gets reprogrammed.
-- 
2.17.1

